mri/operators/gradient/gradient.py

```diff
@@ -0,0 +1,163 @@
+# -*- coding: utf-8 -*-
+##########################################################################
+# pySAP - Copyright (C) CEA, 2017 - 2018
+# Distributed under the terms of the CeCILL-B license, as published by
+# the CEA-CNRS-INRIA. Refer to the LICENSE file or to
+# http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html
+# for details.
+##########################################################################
+
+"""
+This module contains classes for defining gradients.
+"""
+
+# Package import
+from .base import GradBaseMRI
+
+# Third party import
+import numpy as np
+
+
+class GradAnalysis(GradBaseMRI):
+    """ Gradient Analysis class.
+    This class defines the grad operators for:
+            (1/2) * sum(||Ft x - yl||^2_2,l)
+    Attributes
+    ----------
+    data: np.ndarray
+        input data array.
+    fourier_op: an object of class in mri.operators.fourier
+        a Fourier operator from FFT, NonCartesianFFT or Stacked3DNFFT
+    verbose: int, default 0
+        Debug verbosity. Prints debug information during initialization if 1.
+    """
+    def __init__(self, data, fourier_op, verbose=0, **kwargs):
+        if fourier_op.n_coils != 1:
+            data_shape = (fourier_op.n_coils, *fourier_op.shape)
+        else:
+            data_shape = fourier_op.shape
+        super(GradAnalysis, self).__init__(
+            data=data,
+            operator=fourier_op.op,
+            trans_operator=fourier_op.adj_op,
+            shape=data_shape,
+            verbose=verbose,
+            **kwargs)
+        self.fourier_op = fourier_op
+
+
+class GradSynthesis(GradBaseMRI):
+    """ Gradient Synthesis class.
+    This class defines the grad operators for:
+            (1/2) * sum(||Ft Psi_t alpha - yl||^2_2,l)
```

Comment:
Why `Ft` and not `F`?

Reply:
Fourier Transform, I think just that. Legacy..

Reply:
I think it's best to change it to `F`, it is very confusing here, since we might think it is the adjoint operator